### PR TITLE
Verification of CAA records

### PR DESCRIPTION
**Pull request by kumarde**
The code is a lot cleaner and handles everything in the RFC + Errata except for DNSSEC, but I wanted to put this out here first so we can work it out before I add that in. 

From before:

All,

This is an initial implementation of checking CAA records for each domain in a certificate.

There's a lot of code here, so any advice on making this cleaner would be appreciated.

The high level algorithm:

For each DNSName in the SAN, do a CAA lookup (this is complicated in itself. You need to find the closest CAA record, if one exists per domain.) If there's an issue or issuewild tag exist in the answer, check to see if the issuer is actually allowed to issue. If we don't have any data on that issuer, we _skip_ validation because our vantage point doesn't know anything about it.

Thanks!

(ps. This won't build until https://github.com/Censys/censys-definitions/pull/34 lands)

### CODE REVIEW

File: caa.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	zsearch "github.com/censys/censys-definitions/go/censys-definitions"
+	"github.com/miekg/dns"
+	log "github.com/sirupsen/logrus"
+	"github.com/weppos/publicsuffix-go/publicsuffix"
+	"github.com/zmap/zcrypto/x509"
+	"strings"
+	"time"
+)
+
+func runCAAProtobuf(cert *x509.Certificate, parentCertBytes []byte) *zsearch.CAALookup {
```
Comment (zakird) [inline caa.go]: This function name is pretty confusing. I'm not really sure what it's supposed to do.

Comment (kumarde) [reply caa.go]: Updated name.

File: caa.go [hunk 2]
```diff
@@ -48,6 +48,10 @@ func runCAAProtobuf(cert *x509.Certificate, parentCertBytes []byte) *zsearch.CAA
 			fillInCAATagValueProtobuf(&caaTagValue, record)
 			caaRecord.Values = append(caaRecord.Values, &caaTagValue)
 			if record.Tag == "issue" || record.Tag == "issuewild" {
+				if isWildcard && record.Tag == "issue" {
```
Comment (ysf) [inline caa.go]: Actually, if it's a wildcard-domain and no issuewild is set, you fallback on issue. Not skipping it.

Comment (kumarde) [reply caa.go]: good catch.

**Pull request merged**